cEP-0020.md

```diff
@@ -0,0 +1,244 @@
+| -------- | ------------------------------------------------------ |
+| cEP      | 20                                                     |
+| Version  | 1.0                                                    |
+| Title    | Newcomer metrics and Gamification system               |
+| Authors  | Shrikrishna Singh <mailto:krishnasingh.ss30@gmail.com> |
+| Status   | Proposed                                               |
+| Type     | Process                                                |
+
+## Abstract
+
+This cEP describes how a gamification system will automate the process
+of becoming a `Developer` from a `Newcomer`, as a part of this
+[GSoC project](https://summerofcode.withgoogle.com/projects/#5892040252981248).
+
+For the implementation of this process, This project will incorporate a
+gamification system which will allow us to track newcomer's progress and
+assign some points to each of the activity they do. There will be different levels
+based on points threshold and a newcomer will also get some special privileges
+after the compeletion of each level. There will also be different badges on the
+basis of performed activities by the newcomers.
+
+So, every newcomer at coala will have to complete the processes involved in
+the gamification system in order to become a developer at coala or to get the
+**coala Developer** badge.
+
+## Background
+
+### Newcomer Experience
+
+**The problems we have:**
+
+- How to attract more newcomers
+  - How to cope with the increasing amounts of newcomers
+- How to teach and help newcomers to learn faster so
+  we can keep working on coala at a high pace.
+- How to improve newcomer -> dev/maintainer ratio
+- How to increase amount of feedback we get
+  - How to use feedback to improve coala and processes
+
+## Why Gamification?
+
+As coala tries to be a welcoming organization for newcomers, it gives them
+clear pathways and as much direct assistance as possible. However, it is
+already known that newcomers face many barriers while attempting to contribute
+to open source for the first time, and additional barriers while attempting
+to contribute to projects run by the coala organisation.
+
+Some barriers they face include orientation issues that can potentially
+demotivate newcomers from placing their first contribution. On the other
+hand, gamification is widely used to engage and motivate people to
+accomplish tasks and improve their performance.
+
+Therefore, the goal of this project is to use gamification to orient and
+motivate newcomers to overcome onboarding barriers to contribute and engage
+with coala and its community.
+
+## Gamification Elements
+
+There will be four game design elements to help newcomers: Activities, Points,
+Levels and Badges. For each of these elements, we will have a set of rules to
+describe the operation of the gamified environment. i.e.:
+
+### Activities
+
+Let's start by defining some activities which could be tracked:
+
+- Create Issues
+  - Labels?
```

Comment:
>I am fairly confident that the process of awarding a badge atm requires a manual & meticulous check of all activities which contribute to the badge. The system might show that they have finished the necessary activities for a badge, but it isnt yet checked.

Maybe give this privileges to developers?

Reply:
>gitmate label most of the issues as a bug which is not.

I think this will be a big problem for us if we left labeling of issues on gitmate.
As we can never be sure if the label is correct or not.

Maybe we wait for giving the points until the issues is labeled by the maintainer?

Reply:
> Maybe give this privileges to developers?

;-) beyond scope of your project :P